src/iceberg/statistics_file.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#pragma once
+
+/// \file iceberg/statistics_file.h
+/// Statistics file for Iceberg tables.
+
+#include <cstdint>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "iceberg/iceberg_export.h"
+#include "iceberg/util/formattable.h"
+
+namespace iceberg {
+
+/// \brief A metadata about a statistics or indices blob
+struct ICEBERG_EXPORT BlobMetadata : public util::Formattable {
+  /// Type of the blob
+  std::string type;
+  /// ID of the Iceberg table's snapshot the blob was computed from
+  int64_t source_snapshot_id;
+  /// Sequence number of the Iceberg table's snapshot the blob was computed from
+  int64_t source_snapshot_sequence_number;
+  /// Ordered list of fields the blob was calculated from
+  std::vector<int32_t> fields;
+  /// Additional properties of the blob, specific to the blob type
+  std::map<std::string, std::string> properties;
```

Comment:
I think some PRs use unordered_map and some use map - should we be consistent?

Reply:
I have switched to `unordered_map` for now.